umich_catalog_indexing/lib/jobs/zephir_processing/writer.rb

```diff
@@ -0,0 +1,46 @@
+require "zinzout"
+module Jobs
+  module ZephirProcessing
+    class Writer
+      attr_reader :io
+      def initialize(prefix:, scratch_dir: S.scratch_dir, batch_size: S.marc_record_batch_size)
+        @batch_size = batch_size
+        @prefix = prefix
+        @scratch_dir = scratch_dir
+
+        @line_count = 0
+        @file_count = 0
+        _update_current_file_path
```

Comment:
Turn these into functions that take what they need and return a string, so you end up with
@current_file_path = _update_current_file_path(blah blah gblah)